### PR TITLE
tailcfg: add CapabilityFileSharingSendTo to identify FileTargets

### PR DESCRIPTION
**Pull request by maisem**
This adds the inverse to CapabilityFileSharingSend so that senders can
identify who they can Taildrop to.

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -3201,6 +3205,15 @@ func (b *LocalBackend) hasCapFileSharing() bool {
 	return b.capFileSharing
 }
 
+func sliceContains[T comparable](s []T, v T) bool {
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:3208]: https://pkg.go.dev/golang.org/x/exp/slices#Contains

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -1503,6 +1504,9 @@ const (
 	// CapabilityFileSharingSend grants the ability to receive files from a
 	// node that's owned by a different user.
 	CapabilityFileSharingSend = "https://tailscale.com/cap/file-send"
+	// CapabilityFileSharingSendTo grants the ability to send files to a
+	// node that's owned by a different user.
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:1508]: this isn't worded clearly.

It almost sounds backwards.

File: ipn/ipnlocal/local.go (hunk 1 above)
Comment (maisem) [reply ipn/ipnlocal/local.go]: Oh right, done

File: tailcfg/tailcfg.go (hunk 2 above)
Comment (maisem) [reply tailcfg/tailcfg.go]: reworded, PTAL

Comment (maisem) [reply tailcfg/tailcfg.go]: also moved it to the other capability section to be clear that its not in the filter but in`Node.Capabilities`

File: tailcfg/tailcfg.go [hunk 3]
```diff
@@ -1497,6 +1498,9 @@ const (
 	CapabilityAdmin       = "https://tailscale.com/cap/is-admin"
 	CapabilitySSH         = "https://tailscale.com/cap/ssh"         // feature enabled/available
 	CapabilitySSHRuleIn   = "https://tailscale.com/cap/ssh-rule-in" // some SSH rule reach this node
+	// CapabilityFileSharingSendTo grants the current node the ability to send
+	// files to a different node.
+	CapabilityFileSharingSendTo = "https://tailscale.com/cap/file-send-to"
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:1503]: CapabilityFileSharingTarget

**Review by bradfitz — APPROVED** [review]

**Pull request merged**